integrations/apis/flux.weave.works/v1beta1/types.go

```diff
@@ -33,17 +33,40 @@ func (fhr HelmRelease) ResourceID() resource.ID {
 
 // ReleaseName returns the configured release name, or constructs and
 // returns one based on the namespace and name of the HelmRelease.
+// When the HelmRelease's metadata.namespace and spec.targetNamespace
+// differ, both are used in the generated name.
+// This name is used for naming and operating on the release in Helm.
 func (fhr HelmRelease) ReleaseName() string {
-	namespace := fhr.Namespace
-	if namespace == "" {
-		namespace = "default"
+	if fhr.Spec.ReleaseName == "" {
+		namespace := fhr.GetDefaultedNamespace()
+		targetNamespace := fhr.GetTargetNamespace()
+
+		if namespace != targetNamespace {
+			// prefix the releaseName with the administering HelmRelease namespace as well
+			return fmt.Sprintf("%s-%s-%s", namespace, targetNamespace, fhr.Name)
```

Comment:
```suggestion
			return fmt.Sprintf("%s-%s", targetNamespace, fhr.Name)
```
I would keep the original name format to be in line with the Helm2 format.

Reply:
This behavior is only different in the case that there is a parent namespace.
When targetNamespace isn't specified to something different than the HelmRelease ns, the previous behavior is preserved.

This is here to prevent naming conflicts and show (in helm) where the HelmRelease is managed.

Ex:
team1-prod-cache
infra-clusterservices-externaldns
admin-test-privateingress

I'm fine to remove this, but I hope we agree these are helpful auto-generated names. :+1:

Reply:
The release name is used to compose the Kubernetes service names, here is an [example](https://github.com/helm/charts/blob/master/stable/drone/templates/service.yaml#L4). I can see how this would break setups that rely on the Helm v2 convention.

Reply:
`fullname` is a conventional helper template :+1:
The drone chart you linked is using an older version.
It doesn't check if `Release.Name` contains `$name` before appending:
https://github.com/helm/charts/blob/b2425d3/stable/drone/templates/_helpers.tpl#L7-L15

If the release name were `team1-production-drone`,
that fullname would be `team1-production-drone-drone`.

If the release name were `production-drone` (not having the parent namespace prepended), 
it would be `production-drone-drone`.

helm-operator's name convention to prepend the target-namespace is unrelated, additional convention the user must already be aware of.

When composing charts, it's up to users to understand how release names affect resource names. There is no "Helm 2 format" we need to adhere to.
_______________________
Prepending the parent namespace to the release name is good for users because it references what namespace the HelmRelease is in when you do a `helm list`.
This allows you to see all of the releases together, despite them being in different target namespaces.

The one surprising case is modifying the `spec.targetNamespace` of an existing HelmRelease and having the generated name change. Users would have to set the `spec.releaseName` to keep their old release.

This is the only significant trade-off I see in prepending the additional namespace.
If we're not okay with this U/X, I can pull this part of the patch.

I really do see it better for users to have this though.
If you're managing a lot of HelmReleases with targetNamespaces from multiple parent namespaces, this will make it much easier to locate them all from Helm.

Reply:
Ok. Users can change the name in the custom resource if the conventions doesn't suits them.

Reply:
Thanks for hearing me out :+1: